src/scenes/Engagement/LanguageEngagement/Milestone/Milestone.tsx

```diff
@@ -0,0 +1,133 @@
+import { useMutation } from '@apollo/client';
+import { Check, Clear, Edit } from '@mui/icons-material';
+import { CircularProgress, Stack, Tooltip, Typography } from '@mui/material';
+import { useMemo, useState } from 'react';
+import { Form } from 'react-final-form';
+import {
+  LanguageMilestone,
+  LanguageMilestoneLabels,
+  LanguageMilestoneList,
+} from '~/api/schema.graphql';
+import { SelectField } from '~/components/form';
+import { IconButton } from '../../../../components/IconButton';
+import { UpdateLanguageEngagementDocument } from '../../EditEngagement/EditEngagementDialog.graphql';
+import { EngagementMilestoneReachedFragment } from './Milestone.graphql';
+
+interface Props {
+  engagement: EngagementMilestoneReachedFragment;
+}
+
+interface FormShape {
+  milestoneReached: LanguageMilestone;
+}
+
+export const LanguageEngagementMilestone = ({ engagement }: Props) => {
+  const { milestoneReached } = engagement;
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [updateLanguageEngagement] = useMutation(
+    UpdateLanguageEngagementDocument
+  );
+
+  const initialValues = useMemo(
+    (): FormShape => ({
+      milestoneReached: milestoneReached.value || 'Unknown',
+    }),
+    [milestoneReached]
+  );
+
+  if (!milestoneReached.canRead) {
+    return null;
+  }
+
+  return (
+    <Form<FormShape>
+      initialValues={initialValues}
+      onSubmit={async (values, form) => {
+        if (form.getState().dirty) {
+          await updateLanguageEngagement({
+            variables: {
+              input: {
+                engagement: {
+                  id: engagement.id,
+                  milestoneReached: values.milestoneReached,
+                },
+              },
+            },
+          });
+        }
+
+        setIsEditing(false);
+      }}
+      subscription={{ submitting: true }}
+    >
+      {({ handleSubmit, form, submitting }) => (
+        <Stack
+          spacing={1}
+          sx={{ mb: 4 }}
+          component={isEditing ? 'form' : 'div'}
+          onSubmit={handleSubmit}
+        >
+          <Stack direction="row" alignItems="center" spacing={1}>
+            <Typography variant="h3">Milestone</Typography>
```

Comment:
Checking with Tonia on what to label this section.
I'd like
```suggestion
            <Typography variant="h3">Milestone Reached</Typography>
```
but we'll wait and see